amq-message-age/src/main/java/com/signalfx/amq/CollectMessageAge.java

```diff
@@ -0,0 +1,238 @@
+							.build());
+					
+					i.setDatapoint(SignalFxProtocolBuffers.DataPoint
+							.newBuilder()
+							.setMetric("message.age.maximum")
+							.setValue(
+									SignalFxProtocolBuffers.Datum.newBuilder()
+											.setIntValue(messageAge.maximum))
+							.addDimensions(
+                                    SignalFxProtocolBuffers.Dimension
+                                            .newBuilder().setKey("plugin")
+                                            .setValue(PLUGIN))
+							.addDimensions(
+									SignalFxProtocolBuffers.Dimension
+											.newBuilder().setKey("host")
+											.setValue(hostName))
+							.addDimensions(
+									SignalFxProtocolBuffers.Dimension
+											.newBuilder().setKey(queueDimensionName)
+											.setValue(queueNamePrefix + messageAge.queue))
+							.build());
+
+				}
+                
+            } finally {
+                i.close();
+            }
+
+        }
+        
+    }
+    
+    private static class MessageAge{
+    	
+    	public final String queue;
+    	public final long average;
+    	public final long maximum;
+    	
+    	public MessageAge(String queue, long average, long maximum){
+    		this.queue = queue;
+    		this.average = average;
+    		this.maximum = maximum;
+    	}
+    	
+    	@Override
+    	public String toString(){
+    		return "{queue:\"" + queue + "\", average:" + String.valueOf(average) + ", maximum:" + String.valueOf(maximum) + "}";
+    	}
+    	
+    }
+    
+	public static List<MessageAge> collect(String path, String host) {
+
+		List<MessageAge> result = new LinkedList<MessageAge>();
+		
+		try {
+
+			ActiveMQConnectionFactory out = new ActiveMQConnectionFactory(host + "?jms.prefetchPolicy.all=1000");
+			ActiveMQConnection connection;
+			connection = (ActiveMQConnection) out.createConnection();
+
+			connection.start();
+			Session session = connection.createSession(false,Session.AUTO_ACKNOWLEDGE);
+
+			Set<ActiveMQQueue> amqs = connection.getDestinationSource().getQueues();
+			Iterator<ActiveMQQueue> queues = amqs.iterator();
+			
+			while (queues.hasNext()) {
+
+				ActiveMQQueue queue = queues.next();
+				String queueName = queue.getPhysicalName();
+				
+				MessageAge age = collectQueue(path, host, queueName);
+				if(age != null){
+				    result.add(age);
+				}
+
+				
+			}
+			
+			session.close();
+			connection.close();
+
+		} catch (JMSException e) {
+			e.printStackTrace();
+		}
+
+		return result;
+
+	}
+	
+	public static MessageAge collectQueue(String path, String host, String queue) {
+        
+        BufferedReader br = null;
+        
+        try {
+            File file = new File("messages.txt");
+            if(file.exists()){
+                file.delete();
+            }
+            Process p = Runtime.getRuntime().exec(
+                    path + " browse --amqurl " + host + " " + queue + " | grep JMSTimestamp | grep -Eo '[0-9]{1,}' >> messages.txt");
```

Comment:
this writing to a file and then reading into a file is unacceptable. so we're just calling out to another executable within activemq? not sure java is the best language for this, but i guess we're there now (i feel like python would have been better).  At worst, read the stdout of the command and process that without using a file and replace the grepping with regexes in java as well.